share/functions/__fish_man_page.fish

```diff
@@ -1,4 +1,16 @@
 function __fish_man_page
-    man (basename (commandline -po; echo)[1]) ^/dev/null
-    or printf \a
+    # Get all commandline tokens not starting with "-"
+    set -l args (string join \n -- (commandline -po) | grep -v '^-')
+
+    # If there are at least two tokens not starting with "-", the second one might be a subcommand.
+    # Try "man first-second" and fall back to "man first" if that doesn't work out.
+    set -l maincmd (basename $args[1])
```

Comment:
You are right.. I didn't test the empty commandline scenario after adding basename.

I'm adding the check and additionally printing `\a` so that the behavior in case of empty commandline is consistent with other cases (non-existent manpage).